src/wp-includes/cache-compat.php

```diff
@@ -199,3 +199,110 @@ function wp_cache_supports( $feature ) {
 		return false;
 	}
 endif;
+
+if ( ! function_exists( 'wp_cache_get_query_data' ) ) :
+
+	/**
+	 * Retrieves cached query data if valid and unchanged.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @param string $cache_key    The cache key used for storage and retrieval.
+	 * @param string $group        The cache group used for organizing data.
+	 * @param string $last_changed The timestamp of the last modification to the cache group.
+	 * @return mixed|false The cached data if valid, or false if the cache does not exist or is outdated.
+	 */
+	function wp_cache_get_query_data( $cache_key, $group, $last_changed ) {
+		$cache = wp_cache_get( $cache_key, $group );
+
+		if ( ! is_array( $cache ) ) {
+			return false;
+		}
+
+		if ( ! isset( $cache['last_changed'], $cache['data'] ) || $last_changed !== $cache['last_changed'] ) {
+			return false;
+		}
+
+		return $cache['data'];
+	}
+endif;
+
+if ( ! function_exists( 'wp_cache_set_query_data' ) ) :
+	/**
+	 * Stores query-related data in the cache.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @param string $cache_key    The cache key under which to store the data.
+	 * @param mixed  $data         The data to be cached.
+	 * @param string $group        The cache group to which the data belongs.
+	 * @param string $last_changed The timestamp or identifier indicating the last change to the cached data.
+	 * @param int $expire          Optional. When to expire the cache contents, in seconds.
+	 *                             Default 0 (no expiration).
+	 */
+	function wp_cache_set_query_data( $cache_key, $data, $group, $last_changed, $expire = 0 ) {
+		wp_cache_set(
```

Comment:
Many of the existing usages rely on `wp_cache_add`. This shouldn't really make a difference compared to using `wp_cache_set` here, but I want to make sure this difference in function usage is intentional. Are there any potential edge-cases or side effects this change could cause problems with?

Reply:
wp_cache_add was used before as we knew the cache didn't exist before adding. This is case, we are updating existing caches to reuse cache keys. There isn't much difference between add and set. In core, we are pretty incontinence how we use them. I don't see a downside here.